content/sensu-go/5.2/reference/assets.md

```diff
@@ -41,18 +59,120 @@ If the requested asset is not in the local cache, it is downloaded from the asse
 URL. The Sensu backend does not currently provide any storage for assets; they
 are expected to be retrieved over HTTP or HTTPS.
 
-The agent expects that an asset is a `TAR` archive that may optionally be
-GZip'd. Any scripts or executables should be within a `bin/` folder within in
-the archive.
+### Example structure
+{{< highlight shell >}}
+sensu-example-handler_1.0.0_linux_amd64
+├── CHANGELOG.md
+├── LICENSE
+├── README.md
+└── bin
+└── lib
+└── include
+{{< /highlight >}}
 
-The following are injected into the execution context:
+## Sharing an asset on Bonsai
 
-- `{PATH_TO_ASSET}/bin` is injected into the `PATH` environment variable.
-- `{PATH_TO_ASSET}/lib` is injected into the `LD_LIBRARY_PATH` environment
-  variable.
-- `{PATH_TO_ASSET}/include` is injected into the `CPATH` environment variable.
+Share your open-source assets on [Bonsai][16] and connect with the Sensu Community.
+Bonsai supports assets hosted on [GitHub](https://github.com) and released using [GitHub releases](https://help.github.com/articles/about-releases/).
+For more information about creating Sensu Plugins, see the [Sensu Plugin specification][29].
+
+Bonsai requires a [`bonsai.yml` configuration file](#bonsai-yml-specification) in the root directory of your repository that includes the project description, platforms, asset filenames, and SHA-512 checksums.
+For a Bonsai-compatible asset template using Go and [GoReleaser](https://goreleaser.com/), see the [Sensu Go plugin skeleton][28].
```

Comment:
Maybe call out in some fashion how non-Go assets would work or link to that content if already created.

Reply:
AFAICT, the bonsai web app receives webhook events from github that tell it to update the index entry for the corresponding asset, an action which I think is initiated by the creation of the release itself, not the files associated, so there's no direct dependency on using goreleaser.

my understanding is that goreleaser's function is to attach (upload) assets built by travis to [GitHub releases](https://help.github.com/articles/about-releases/).

anyone not using goreleaser can manually create a release on github and upload files or, use some other tool to automate these tasks.

Reply:
So if we want to say something like "Assets can be written in any language as long as..." How does that sentence end?